src/item_factory.cpp

```diff
@@ -2337,6 +2349,36 @@ void Item_factory::set_qualities_from_json( const JsonObject &jo, const std::str
     }
 }
 
+void Item_factory::extend_qualities_from_json( const JsonObject &jo, const std::string &member,
+        itype &def )
+{
+    for( JsonArray curr : jo.get_array( member ) ) {
+        const auto quali = std::pair<quality_id, int>( quality_id( curr.get_string( 0 ) ),
```

Comment:
```suggestion
        def.qualities[quality_id( curr.get_string( 0 ) )] = curr.get_int( 1 )
```
This can replace the whole content of the loop body.